pkg/apply/apply.go

```diff
@@ -63,3 +76,38 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 
 	return nil
 }
+
+// ApplyObjects it applies a list of desired objects after merging them.
+func ApplyObjects(ctx context.Context, client k8sclient.Client, objs []*uns.Unstructured) error {
+
+	var lastObj, existing *uns.Unstructured = nil, nil
+	var objDesc string = ""
+	var err error = nil
+
+	for _, obj := range objs {
+		existing, objDesc, err = findOrCreateObject(ctx, client, obj)
```

Comment:
I believe this can be called out of the loop with the first item of the slice, it will save us some gets